### ISSUE TITLE
Repo: "Package failed to publish: ... E404 Not found" publishing canary versions

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#8086]
### Suggestion

This has happened in the last few commits to `main`, including -> https://github.com/typescript-eslint/typescript-eslint/actions/runs/7196433948/job/19601821500:

```plaintext
...
lerna verb packed packages/type-utils
lerna verb packed packages/rule-tester
lerna verb pack-directory packages/eslint-plugin
lerna verb packed packages/eslint-plugin
lerna verb publish @typescript-eslint/types
lerna WARN notice Package failed to publish: @typescript-eslint/types
lerna ERR! E404 Not found
lerna ERR! errno "undefined" is not a valid exit code - exiting with code 1
```

**Comment by auvred** [comment]
Maybe we can try to pass `--loglevel=silly` to the lerna? it will show more readable error
https://github.com/typescript-eslint/typescript-eslint/blob/fca72762772833d1d696d9860c535c294ea24245/.github/workflows/ci.yml#L295

When I tried to reproduce it locally with verdaccio, it threw the following (more verbose than just `E404 Not found`):

<details>

<summary>
Error log
</summary>

```shell
lerna verb publish @typescript-eslint/visitor-keys
lerna sill HttpErrorGeneral: 404 Not Found - PUT https://registry.npmjs.org/@typescript-eslint%2fvisitor-keys - Not found
lerna sill     at /home/typescript-eslint/node_modules/npm-registry-fetch/lib/check-response.js:95:15
lerna sill     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
lerna sill     at async publish (/home/typescript-eslint/node_modules/libnpmpublish/lib/publish.js:54:17)
lerna sill     at async /home/typescript-eslint/node_modules/p-pipe/index.js:12:19
lerna sill     at async run (/home/typescript-eslint/node_modules/p-queue/dist/index.js:163:29)
lerna sill  HttpErrorGeneral: 404 Not Found - PUT https://registry.npmjs.org/@typescript-eslint%2fvisitor-keys - Not found
lerna sill     at /home/typescript-eslint/node_modules/npm-registry-fetch/lib/check-response.js:95:15
lerna sill     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
lerna sill     at async publish (/home/typescript-eslint/node_modules/libnpmpublish/lib/publish.js:54:17)
lerna sill     at async /home/typescript-eslint/node_modules/p-pipe/index.js:12:19
lerna sill     at async run (/home/typescript-eslint/node_modules/p-queue/dist/index.js:163:29) {
lerna sill   headers: [Object: null prototype] {
lerna sill     date: [ 'Fri, 22 Dec 2023 11:46:46 GMT' ],
lerna sill     'content-type': [ 'application/json' ],
lerna sill     'content-length': [ '21' ],
lerna sill     connection: [ 'keep-alive' ],
lerna sill     'cf-ray': [ '8398204d4a8b0061-DME' ],
lerna sill     vary: [ 'Accept-Encoding' ],
lerna sill     server: [ 'cloudflare' ],
lerna sill     'x-fetch-attempts': [ '1' ]
lerna sill   },
lerna sill   statusCode: 404,
lerna sill   code: 'E404',
lerna sill   method: 'PUT',
lerna sill   uri: 'https://registry.npmjs.org/@typescript-eslint%2fvisitor-keys',
lerna sill   body: { error: 'Not found' },
lerna sill   pkgid: '@typescript-eslint/visitor-keys@6.15.1-alpha.1'
lerna sill }
lerna WARN notice Package failed to publish: @typescript-eslint/visitor-keys
lerna ERR! E404 Not found
lerna ERR! errno "undefined" is not a valid exit code - exiting with code 1
```

</details>

P.S
After I set registry url to the `localhost`, everything published fine. I don't think it's broken in GH Actions due to wrong registry url, but more detailed logs might help us better understand what's going on

**Comment by JamesHenry** [comment]
Publishing canaries from main is working again now. It turns out it was as simple as the NPM access token having expired. NPM, unlike GitHub, did not provide any notification about this.

So if you see 404s errors from publishing in the future, the first thing we should check is the expiry of the NPM token (I've also set a personal reminder for the expiry of the updated token to try and proactively avoid it next time around)
